linalg/amgxsolver.hpp

```diff
@@ -162,6 +162,11 @@ class AmgXSolver : public Solver
    */
    void DefaultParameters(const AMGX_MODE amgxMode_, const bool verbose);
 
+   // Configure as AmgXSolver::PRECONDITIONER or AmgXSolver::SOLVER.
+   // When configuring as a preconditioner, the check for convergence
+   // or divergence will be skipped after applying Mult.
+   void ConfigureAs(const AMGX_MODE amgxMode_);
```

Comment:
In my view this name is wrong, because calling this does not in fact reconfigure the object. The name makes me expect that if I construct with `AmgXSolver(PRECONDITIONER)` and then call `.ConfigureAs(SOLVER)` it will change things, but that's not the case. I suggest adding a `check_diverged` flag or something similar and change this method to `SetConvergenceCheck()`. Then `DefaultParameters()` can still set the new flag based on the mode. Does that make sense?

Reply:
Yes! That is more clear, I will update it accordingly.